### DISCUSSION TITLE
Keep the self registration as an option for users within a specified domain & option to force first name/last name entries

### DISCUSSION THREAD

**Discussion by tbendien** [bionic-gpt/bionic-gpt#240] [Q&A]
As you develop new user registration methods, could you keep the option for a sysadmin to enable/allow self registration with a specific email domain please?

For example, if I am putting the system into a workgroup, then it would be really helpful if the Bionic sysadmin can set a rule to say that all users with a certain FQDN email address are allowed to self register on the system.

I am planning to run a closed user group that all belong to a certain organization, and I want to allow them to self register.

I'd also like the option to force a user to put in a first and last name at registration time. Obviously they can make something up, however, as things progress, I want to be able to run analytics and searches based on user names as well as email addresses for the chat history. It will make it a bit easier to see who is doing what on the system if our end user policy for the closed workgroup is that they have to put their real name into the system so it's easier to audit chat logs.

Let's say I am rolling out the system to a school, I want to be able to run SQL that correlates a user with a school email domain with their self provided first & last name e.g. do a join query with a separate school database to make sure the user has given their real name as is associated with their school email.

A school system I am working with uses an anonymous email address format, so it is not possible to identify a student based on their school email. The format of their school email is: 123456@school.edu

**Comment by ianpurton** [comment]
The new demo version of Bionic will use [Keycloak](https://www.keycloak.org/)

Bionic will support [OpenID Connect (OIDC)](https://en.wikipedia.org/wiki/OpenID#OpenID_Connect_(OIDC))

You should be able to use any IAM system that support OIDC.  

Not sure if keycloak supports the features you want.

![image](https://github.com/bionic-gpt/bionic-gpt/assets/36966/ca6beb89-aec4-41f6-b14f-a03683efff41)
